plugins/MauticCrmBundle/Integration/SalesforceIntegration.php

```diff
@@ -1751,7 +1762,7 @@ protected function getMauticContactsToUpdate(
             if (!empty($lead['email'])) {
                 $lead                                               = $this->getCompoundMauticFields($lead);
                 $key                                                = $this->getSyncKey($lead['email']);
-                $trackedContacts[$lead['integration_entity']][$key] = $lead['id'];
+                $trackedContacts[$lead['integration_entity']][$key] = $lead['internal_entity_id'];
```

Comment:
Why was this changed? It should be the ID of the IntegrationEntity as it's used to obtain references to it in other places. https://github.com/mautic/mautic/blob/3fc67864148828ec4c2f78676be149a276e8e5ac/plugins/MauticCrmBundle/Integration/SalesforceIntegration.php#L2321